cmd/soroban-cli/src/config/secret.rs

```diff
@@ -39,19 +41,26 @@ pub struct Args {
     /// Add using 12 word seed phrase to generate `secret_key`
     #[arg(long, conflicts_with = "secret_key")]
     pub seed_phrase: bool,
+
+    /// Add a public key, ed25519, or muxed account, e.g. G1.., M2..
+    #[arg(long, conflicts_with = "seed_phrase", conflicts_with = "secret_key")]
+    pub public_key: Option<String>,
 }
 
 impl Args {
-    pub fn read_secret(&self) -> Result<Secret, Error> {
+    pub fn read_key(&self) -> Result<Key, key::Error> {
```

Comment:
Here it is just for parsing the Arg struct. So it only should be used with the `keys add` command. Though then maybe it should just go into that command? Since there is no reuse of this function.

Reply:
I kind of like the idea of moving it to the `add` command, since that is where it's being used. But not merge blocking!